File: app/src/main/java/com/szeptun/shoppinglist/domain/RemoveProduct.java
package com.szeptun.shoppinglist.domain;

import com.szeptun.shoppinglist.dataaccess.repository.ShoppingListRepository;
import com.szeptun.shoppinglist.entity.Product;

import javax.inject.Inject;

import io.reactivex.Completable;

public class RemoveProduct {

    @Inject
    RemoveProduct(){}

    @Inject
    ShoppingListRepository repository;

    public Completable execute(Product product){
        return repository.removeProduct(product);
    }
}

File: app/src/main/java/com/szeptun/shoppinglist/domain/RemoveProduct.kt
package com.szeptun.shoppinglist.domain

import com.szeptun.shoppinglist.dataaccess.repository.ShoppingListRepository
import com.szeptun.shoppinglist.entity.Product
import io.reactivex.Completable
import javax.inject.Inject

class RemoveProduct @Inject constructor(private val repository: ShoppingListRepository) {

    fun execute(product: Product): Completable =
        repository.removeProduct(product)
}